src/Dianoga/Invokers/MediaCacheAsync/OptimizingMediaCache.cs

```diff
@@ -50,66 +53,80 @@ public override bool AddStream(Media media, MediaOptions options, MediaStream st
 
 			// Sitecore will use this to stream the media while we persist
 			cachedStream = stream;
-
-			// make a copy of the stream to use temporarily while we async persist
-			var copyStream = new MemoryStream();
-			stream.CopyTo(copyStream);
-			copyStream.Seek(0, SeekOrigin.Begin);
-
-			var copiedMediaStream = new MediaStream(copyStream, stream.Extension, stream.MediaItem);
-
+			
 			// we store the site context because on the background thread: without the Sitecore context saved (on a worker thread), that disables the media cache
 			var currentSite = Context.Site;
 
-			ThreadPool.QueueUserWorkItem(state =>
-			{
-				var mediaPath = stream.MediaItem.MediaPath;
+			_actionBlock.Post(() => Optimize(currentSite, stream.MediaItem, media, options));
+
+			return true;
+		}
 
-				try
+		private void Optimize(SiteContext currentSite, MediaItem mediaItem, Media media, MediaOptions options)
+		{
+            MediaStream originalMediaStream = null;
+			MediaStream backupMediaStream = null;
+			MediaStream optimizedMediaStream = null;
+
+            try
+            {
+				//get stream from mediaItem to reduce memory usage
```

Comment:
Can you reformat the whitespace please :)

Reply:
definitely, done!